docs/source/dev_guide/project-standards.rst

```diff
@@ -0,0 +1,190 @@
+
+Source: https://gist.github.com/jbenet/ee6c9ac48068889b0912
+
+Pre-commit
+~~~~~~~~~~
+The repository uses the ``pre-commit`` package to manage pre-commit hooks.
+These hooks help enforce QA standards and identify simple issues
+at the commit level before submitting code reviews.
+
+.. figure:: pre-commit-flow.svg
+   :alt: Pre-commit Flow Diagram
+
+   ``pre-commit`` Flow
+
+Helpful ``pre-commit`` Commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install into your cloned repo ::
+
+    conda activate zstash_dev
+    pre-commit install
+
+Automatically run all pre-commit hooks (just commit) ::
+
+    # Tip: If there is an issue with pre-commit, you can bypass with the `--no-verify` flag. Please do NOT use this on a regular basis.
+    git commit -m '...'
+
+.. figure:: ../_static/pre-commit-passing.png
+   :alt: pre-commit Output
+
+   ``pre-commit`` Output
+
+Manually run all pre-commit hooks ::
+
+    pre-commit run --all-files
+
+Run individual hook ::
+
+    # Available hook ids: trailing-whitespace, end-of-file-fixer, check-yaml, black, isort, flake8, mypy
+    pre-commit run <hook_id>
+
+Squash and Rebase Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before you merge a support branch back into ``master``, the branch is typically
+squashed down to a single* buildable commit, and then rebased on top of the main repo's ``master`` branch.
+
+\* *In some cases, it might be logical to have multiple squashed commits, as long as each commit passes the CI/CD build*
+
+Why squash and rebase commits?
+
+- Ensures build passes from the commit
+- Cleans up Git history for easy navigation
+- Makes collaboration and review process more efficient
+- Makes handling conflicts from rebasing simple since you only have to deal with conflicted commits
+- Makes ``git bisect`` easier and more effective to use. For example, it will show the exact commit that introduced a bug since the commit contains a relatively small changeset. On the otherhand, merge commits make it much harder since it includes a large changeset.
+
+
+How to squash and rebase commits
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Assuming that you followed :ref:`"(b) Development Environment" <dev-env>`:
+
+1. Sync ``master`` with the main repo's ``master`` ::
+
+    git checkout master
+    git rebase <upstream-origin>/master
+    git push -f <fork-origin> master
+
+2. Squash commits::
+
+    git rebase -i [SHA]
+
+    # OR
+
+    git rebase -i HEAD~[NUMBER OF COMMITS]
+
+3. Make sure your squashed commit messages are refined
+
+4. Get the SHA of the commit OR number of commits to rebase to ::
+
+    git log --graph --decorate --pretty=oneline --abbrev-commit
+
+5. Rebase branch onto ``master`` ::
+
+    git checkout <branch-name>
+    git rebase master # If you have merge conflicts, it may be easier to `git rebase --abort` and first do step 4 with SHA of the last commit before your work
+    git push -f <fork-origin> <branch-name>
+
+6. Force push to remote branch ::
```

Comment:
Might be helpful to note `You have to push force push because the rebase rewrites the commit SHAs`.